sam/models/TimeseriesMLP.py

```diff
@@ -3,103 +3,18 @@
-            This will be used to do the preprocessing before calling explainer.shap_values
-        """
-        self.explainer = explainer
-
-        # Create a proxy model that can call only 3 attributes we need
-        class SamProxyModel:
-            fit = None
-            use_y_as_feature = model.use_y_as_feature
-            feature_names_ = model.get_feature_names()
-            preprocess_predict = SamQuantileMLP.preprocess_predict
-
-        self.model = SamProxyModel()
-        # Trick sklearn into thinking this is a fitted variable
-        self.model.feature_engineer_ = model.feature_engineer_
-        # Will likely be somewhere around 0
-        self.expected_value = explainer.expected_value
-
-    def shap_values(self, X: pd.DataFrame, y: pd.Series = None, *args, **kwargs) -> np.array:
-        """
-        Imitates explainer.shap_values, but combined with the preprocessing from the model.
-        Returns a similar format as a regular shap explainer: a list of numpy arrays, one
-        for each output of the model.
-
-        Parameters
-        ----------
-        X: pd.DataFrame
-            The dataframe used to 'train' the explainer
-        y: pd.Series, optional (default=None)
-            Target data used to 'train' the explainer.
-        """
-        X_transformed = self.model.preprocess_predict(X, y, dropna=False)
-        return self.explainer.shap_values(X_transformed, *args, **kwargs)
-
-    def attributions(self, X: pd.DataFrame, y: pd.Series = None, *args, **kwargs) -> np.array:
-        """
-        Imitates explainer.attributions, which by default just mirrors shap_values
-
-        Parameters
-        ----------
-        X: pd.DataFrame
-            The dataframe used to 'train' the explainer
-        y: pd.Series, optional (default=None)
-            Target data used to 'train' the explainer.
-        """
-        return self.shap_values(X, y, *args, **kwargs)
-
-    def test_values(self, X: pd.DataFrame, y: pd.Series = None) -> pd.DataFrame:
-        """
-        Only the preprocessing from the model, without the shap values.
-        Returns a pandas dataframe with the actual values used for the explaining
-        Can be used to better interpret the numpy array that is outputted by shap_values.
-        For example, if shap_values outputs a 5x20 numpy array, that means you explained 5 objects
-        with 20 features. This function will then return a 5x20 pandas dataframe.
-
-        Parameters
-        ----------
-        X: pd.DataFrame
-            The dataframe used to 'train' the explainer
-        y: pd.Series, optional (default=None)
-            Target data used to 'train' the explainer.
-        """
-        X_transformed = self.model.preprocess_predict(X, y, dropna=False)
-        return pd.DataFrame(X_transformed, columns=self.model.feature_names_, index=X.index)
-
-
-class SamQuantileMLP(BaseTimeseriesRegressor):
+class TimeseriesMLP(BaseTimeseriesRegressor):
```

Comment:
I still don't know what to think of the name. I doesn't say anything about quantiles, while that's the unique thing in my opinion